test/wh_test_crypto.c

```diff
@@ -438,9 +491,74 @@ int whTest_CryptoClientConfig(whClientConfig* config)
     }
     if (XMEMCMP(sharedOne, sharedTwo, outLen) != 0) {
         WH_ERROR_PRINT("CURVE25519 shared secrets don't match\n");
+        ret = -1;
+        goto exit;
     }
-
-
+    /* test cmac */
+    if((ret = wc_InitCmac_ex(cmac, knownCmacKey, sizeof(knownCmacKey), WC_CMAC_AES, NULL, NULL, WOLFHSM_DEV_ID)) != 0) {
+        WH_ERROR_PRINT("Failed to wc_InitCmac_ex %d\n", ret);
+        goto exit;
+    }
+    if((ret = wc_CmacUpdate(cmac, (byte*)knownCmacMessage, sizeof(knownCmacMessage))) != 0) {
+        WH_ERROR_PRINT("Failed to wc_CmacUpdate %d\n", ret);
+        goto exit;
+    }
+    outLen = sizeof(knownCmacTag);
+    if((ret = wc_CmacFinal(cmac, (byte*)cipherText, &outLen)) != 0) {
+        WH_ERROR_PRINT("Failed to wc_CmacFinal %d\n", ret);
+        goto exit;
+    }
+    if (memcmp(knownCmacTag, cipherText, sizeof(knownCmacTag)) != 0) {
+        WH_ERROR_PRINT("CMAC FAILED KNOWN ANSWER TEST\n");
+        ret = -1;
+        goto exit;
+    }
+    if((ret = wc_AesCmacVerify_ex(cmac, (byte*)knownCmacTag, sizeof(knownCmacTag), (byte*)knownCmacMessage, sizeof(knownCmacMessage), knownCmacKey, sizeof(knownCmacKey), NULL, WOLFHSM_DEV_ID)) != 0) {
+        WH_ERROR_PRINT("Failed to wc_AesCmacVerify_ex %d\n", ret);
+        goto exit;
+    }
+    /* test oneshot with pre-cached key */
+    if ((ret = wh_Client_KeyCache(client, 0, labelStart, sizeof(labelStart), knownCmacKey, sizeof(knownCmacKey), &keyId)) != 0) {
+        WH_ERROR_PRINT("Failed to wh_Client_KeyCache %d\n", ret);
+        goto exit;
+    }
+    outLen = sizeof(knownCmacTag);
+    if((ret = wh_Client_AesCmacGenerate(cmac, (byte*)cipherText, &outLen, (byte*)knownCmacMessage, sizeof(knownCmacMessage), keyId, NULL)) != 0) {
+        WH_ERROR_PRINT("Failed to wh_Client_AesCmacGenerate %d\n", ret);
+        goto exit;
+    }
+    if (memcmp(knownCmacTag, cipherText, sizeof(knownCmacTag)) == 0)
+        printf("CMAC SUCCESS\n");
+    else {
+        WH_ERROR_PRINT("CMAC FAILED KNOWN ANSWER TEST\n");
+        ret = -1;
```

Comment:
Should we return early too?
```suggestion
        ret = -1;
        goto exit;
```